static/app/components/forms/selectOption.tsx

```diff
@@ -0,0 +1,143 @@
+import {components as selectComponents} from 'react-select';
+import styled from '@emotion/styled';
+
+import {IconCheckmark} from 'sentry/icons';
+import space from 'sentry/styles/space';
+
+type Props = React.ComponentProps<typeof selectComponents.Option>;
+
+const Option = (props: Props) => {
+  return (
+    <Wrap>
+      <selectComponents.Option {...props}>
+        <InnerWrap isFocused={props.isFocused}>
+          <Indent>
+            <CheckWrap isMulti={props.isMulti} isSelected={props.isSelected}>
+              {props.isSelected && (
+                <IconCheckmark
+                  size={props.isMulti ? 'xs' : 'sm'}
+                  color={props.isMulti ? 'white' : undefined}
+                />
+              )}
+            </CheckWrap>
+            {props.data.leadingItems && (
+              <LeadingItems>{props.data.leadingItems}</LeadingItems>
+            )}
+          </Indent>
+          <ContentWrap
+            isFocused={props.isFocused}
+            showDividers={props.selectProps.showDividers}
+          >
+            <div>
+              <Label>{props.label}</Label>
+              {props.data.details && <Details>{props.data.details}</Details>}
+            </div>
+            {props.data.leadingItems && (
+              <TrailingItems>{props.data.trailingItems}</TrailingItems>
+            )}
+          </ContentWrap>
+        </InnerWrap>
+      </selectComponents.Option>
+    </Wrap>
+  );
+};
+
+export default Option;
+
+const Wrap = styled('div')``;
+
+const InnerWrap = styled('div')<{isFocused: boolean}>`
+  display: flex;
+  padding: 0 ${space(1)};
+  border-radius: ${p => p.theme.borderRadius};
+  box-sizing: border-box;
+
+  ${p => p.isFocused && `background: ${p.theme.hover};`}
+`;
+
+const Indent = styled('div')`
+  display: flex;
+  justify-content: center;
+  gap: ${space(1)};
```

Comment:
We should officially just start using `gap` now and translate all of our `grid-gap` to just `gap`, and switch to flex when possible 🤔